src/app/screens/Liquid/ConfirmSignPset.tsx

```diff
@@ -0,0 +1,230 @@
+
+  useEffect(() => {
+    (async () => {
+      if (!pset) throw new Error("pset is undefined");
+      const account = await api.getAccount();
+      const liquidNetwork = toLiquidNetworkName(account.bitcoinNetwork);
+      const preview = await api.liquid.getPsetPreview(pset, liquidNetwork);
+      setLiquidNetwork(liquidNetwork);
+      setPreview(preview);
+
+      const esploraClient = Esplora.fromNetwork(liquidNetwork);
+      const registry = await fetchAssetRegistry(
+        esploraClient,
+        preview,
+        console.error // do not throw, just log errors
+      );
+
+      setAssetRegistry(registry);
+    })();
+  }, [pset]);
+
+  async function confirm() {
+    try {
+      setLoading(true);
+      const response = await msg.request(
+        "liquid/signPset",
+        { pset: pset, network: liquidNetwork },
+        { origin }
+      );
+      msg.reply(response);
+      setSuccessMessage(tCommon("success"));
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error) toast.error(`${tCommon("error")}: ${e.message}`);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  function reject(e: React.MouseEvent<HTMLAnchorElement>) {
+    e.preventDefault();
+    msg.error(USER_REJECTED_ERROR);
+  }
+
+  function close(e: React.MouseEvent<HTMLButtonElement>) {
+    if (navState.isPrompt) {
+      window.close();
+    } else {
+      e.preventDefault();
+      navigate(-1);
+    }
+  }
+
+  function toggleShowAddresses() {
+    setShowAddresses((current) => !current);
+  }
+  function toggleShowHex() {
+    setShowHex((current) => !current);
+  }
+
+  if (!preview) {
+    return <Loading />;
+  }
+
+  return (
+    <div className="h-full flex flex-col overflow-y-auto no-scrollbar">
+      <ScreenHeader title={t("title")} />
+      {!successMessage ? (
+        <Container justifyBetween maxWidth="sm">
+          <div className="flex flex-col gap-4 mb-4">
+            <PublisherCard
+              title={origin.name}
+              image={origin.icon}
+              url={origin.host}
+            />
+            <div className="rounded-md font-medium p-4 text-sm text-orange-700 bg-orange-50 dark:text-orange-400 dark:bg-orange-900">
+              {t("warning")}
+            </div>
+            <div className="p-4 shadow bg-white dark:bg-surface-02dp rounded-lg overflow-hidden flex flex-col gap-4">
+              <h2 className="font-medium dark:text-white">
+                {t("allow_sign", { host: origin.host })}
+              </h2>
+              <div className="flex gap-2">
+                <Hyperlink onClick={toggleShowAddresses}>
+                  {showAddresses ? t("hide_addresses") : t("view_addresses")}
+                </Hyperlink>
+                <span>{"•"}</span>
+                <Hyperlink onClick={toggleShowHex}>
+                  {showHex ? t("hide_hex") : t("view_hex")}
+                </Hyperlink>
+              </div>
+
+              {showAddresses && (
```

Comment:
@louisinger should we show the fee somewhere on the confirmation dialog? it can be calculated just subtracting the input from the output?

Reply:
Because of confidential feature, the subtract trick won't work. That's why Liquid txs have a "fee output" which is an output with a value (the fees amount) and an empty script. It should be easy to get that value in `getPsetPreview`. 

Beside that, not sure if we really need that feature. FYI Marina does not show the fee amount (because most of the app/explorers show them)

Reply:
@louisinger I wonder if it would be good to have it in the PsetPreview since Alby should be the source of truth (another app could get the fee wrong and users might not know?). But this could be a future improvement.

Reply:
@rolznz as you prefer, let me know, I can open a PR to update the PsetPreview :grin: 

> Alby should be the source of truth

To be honest, because Alby does not handle unblinding, this can't be true. the values/assets in the preview are computed from data in clear set up by the app requesting signing (in case of confidential outputs). In other words, _Alby can't unblind confidential inputs/outputs to check if the app isn't lying_. Maybe user should be aware of this (via a warning or something in the popup) ?

Reply:
@louisinger The signing screen is to make sure the user knows exactly what they are signing before confirming signing the transaction. Alby is the sender and generates the blinding key, shouldn't it know exactly what it is signing? sorry, I do not know enough about Liquid here, but it seems wrong to me if the user cannot verify what they are signing.

@bumi @reneaaron do you have any thoughts here?

Reply:
> shouldn't it know exactly what it is signing?

to do this, I would need https://github.com/vulpemventures/secp256k1-zkp module (in order to verify the ZKP proofs). So it means another dependencie (as far as I know `noble/curves` does not handle ZKP, right?)

Reply:
@louisinger we chatted with @bumi and @reneaaron, we think there is no need to do this now and users should have some level of trust with the site they are using. We can re-review if there is more usage. Thanks!